lib/model/message.dart

```diff
@@ -35,6 +35,34 @@ mixin MessageStore {
   /// All [Message] objects in the resulting list will be present in
   /// [this.messages].
   void reconcileMessages(List<Message> messages);
+
+  /// Whether the current edit request, if any, has failed.
+  ///
+  /// Will be null if there is no current edit request
+  /// or the message was deleted.
+  /// Will be false if the current request hasn't failed
+  /// and the update-message event hasn't arrived.
+  bool? getEditMessageErrorStatus(int messageId);
+
+  /// Edits a message's content.
+  ///
+  /// See also:
+  ///   * [getEditMessageErrorStatus]
+  ///   * [takeFailedMessageEdit]
+  void editMessage({required int messageId, required String content});
```

Comment:
I think we should document the condition when you should not call `editMessage` (like we do for `takeFailedMessageEdit`), i.e., when there is a current edit message request.